content/04-guides/01-database-workflows/08-sql-views.mdx

```diff
@@ -0,0 +1,378 @@
+
+   ```sql
+   CREATE TABLE "public"."User" (
+     id SERIAL PRIMARY KEY,
+     email TEXT UNIQUE
+     name TEXT
+   );
+
+   CREATE TABLE "public"."Post" (
+     id SERIAL PRIMARY KEY,
+     title TEXT,
+     content TEXT,
+     published BOOLEAN,
+     "authorId" INTEGER,
+     CONSTRAINT author FOREIGN KEY ("authorId") REFERENCES "public"."User" (id)
+   );
+   ```
+
+2. Run the SQL statement against your database to create the two tables:
+
+   ```
+   psql SqlViews < sql-views-tables.sql
+   ```
+
+3. Run the following command to validate that the tables were created:
+
+   ```
+   psql -d SqlViews -c "\dt"
+   ```
+
+You just created two tables named `User` and `Post` in the database. The `Post` table references the `User` table via the foreign key defined on the `authorId` column.
+
+## 3. Create a view named 'Draft'
+
+In this section you will create a view named Drafts that returns the post title and author email of all posts that have not been published. To create a view:
+
+1. Create a new file called `sql-views-draft.sql` and add the following code to it:
+
+   ```sql
+    CREATE VIEW "Draft" AS
+        SELECT "published", "title", "email", "Post"."id"
+        FROM "Post", "User"
+        WHERE "published" = false AND "Post"."authorId" = "User"."id";
+   ```
+
+2. Run the SQL statement against your database to create the view:
+
+   ```
+   psql SqlViews < sql-views-draft.sql
+   ```
+
+3. Run the following command to validate that the view was created:
+
+   ```
+   psql -d SqlViews -c "\dv"
+   ```
+
+   You should see the following list of views:
+
+   ```
+         List of relations
+    Schema |  Name  | Type |  Owner
+    --------+--------+------+----------
+    public | Draft  | view | postgres
+   ```
+
+## 4. Introspect your database with Prisma
+
+In this section you'll introspect your database to generate the Prisma models for the tables that you created.
+
+> **Note**: You will manually add the Draft view to the Prisma schema in a later step.
+
+1. Set up a new Node.js project and add the `prisma` CLI as a development dependency:
+
+   ```
+   npm init -y
+   npm install prisma --save-dev
+   ```
+
+2. Create a new file called `schema.prisma` and add the following code to it:
+
+   ```prisma
+   datasource postgresql {
+       provider = "postgresql"
+       url      = env("DATABASE_URL")
+   }
+   ```
+
+3. In order to introspect your database, you need to tell Prisma how to connect to it. You do so by configuring a `datasource` in your Prisma schema. Create a new file called `.env` and set your database connection URL as the `DATABASE_URL` environment variable:
+
+   ```
+   DATABASE_URL=postgresql://__USER__:__PASSWORD__@__HOST__:__PORT__/CascadingDeletesDemo
+   ```
+
+   In the above code snippet, you need to replace the uppercase placeholders with your own connection details. For example, if your database is running locally it could look like this:
+
+   ```
```

Comment:
the code block seems broken 🤔 (I didn't run locally, but the formatting here in the PR seems a bit strange)